pkg/controllers/management/etcdbackup/etcdbackup.go

```diff
@@ -291,7 +305,65 @@ func NewBackupObject(cluster *v3.Cluster, manual bool) *v3.EtcdBackup {
 			ClusterID: cluster.Name,
 			Manual:    manual,
 		},
+		Status: v3.EtcdBackupStatus{
+			KubernetesVersion: cluster.Spec.RancherKubernetesEngineConfig.Version,
+			ClusterObject:     compressedCluster,
+		},
+	}, nil
+}
+
+func CompressCluster(cluster *v3.Cluster) (string, error) {
+	jsonCluster, err := json.Marshal(cluster)
+	if err != nil {
+		return "", err
+	}
+
+	var gzCluster bytes.Buffer
+	gz := gzip.NewWriter(&gzCluster)
+
+	_, err = gz.Write([]byte(jsonCluster))
+	if err != nil {
+		return "", err
+	}
+
+	if err = gz.Flush(); err != nil {
+		return "", err
+	}
```

Comment:
It is safer to call gz.Flush() in a defer func() to ensure flush is called even if there is an error in Write.